### PR TITLE
Poking at blobs

### PR DESCRIPTION
**Pull request by ghost**
Blob core cost now scales with the number of cores in existence (Currently set to +50 for each core in existence besides your own). To subside this, each node a core **places** _not takes over_ adds 10 additional blob point capacity, so a blob would have to make more nodes to be able to make more cores.

Blobs attacking supermatter now actually call explode on supermatter, rather than just making them disappear from existence.

Admins are now informed of who's saying what when it comes to blob telepathy. No more WGW

blob spores now inherit their factory DMI, so you can toss those in to a themed DMI

2 new blob skins - Honkscape (Courtesy of blithering)
blob_honk

and AME (Inspired by a bus celt did that one time)
blob_ame

More blob tweaks are followed if suggested

:cl:
- rscadd: Blob core cost now scales with the number of active blob cores in existence
- rscadd: Maximum blob points now scale with the number of nodes a blob core has CREATED
- rscadd: A few alternate blob designs are now available for admins to change blobs to
- rscadd: Blob spores now inherit the icon_state of the factory that creates them, opening up further customization options for blob
- rscadd: Admins are now given a proper log of who said what as a blob who is using telepathy
- rscdel: Removes the pulsing animation from 64x64 blobs, should reduce lag when blobs are active
- rscadd: Attacking supermatter as a blob now blows it the fuck up, rather than wiping it from existence

### CODE REVIEW

File: code/game/gamemodes/blob/powers.dm [hunk 1]
```diff
@@ -123,7 +123,16 @@
 		to_chat(src, "There is another core blob nearby, move more than 15 tiles away from it!")
 		return
 
-	if(!can_buy(100))
+	var/cost = 100
+	var/number_of_cores
+
+	for(var/obj/effect/blob/core/C in world)
```
Comment (clusterfack) [inline code/game/gamemodes/blob/powers.dm]: Don't loop through world

**Comment by clusterfack** [comment]
I don't see how an admin could fail to see it since it sends to everyone in world

**Comment by clusterfack** [comment]
Also I think the cost should scale, (since I'm the one who suggested it initially), but I think it should scale at a cost different than the base cost.

Like 100 base points, scales up by 25/30, node scales up by 15 for example? Here its 100, 200, 300? You'll never get to that point if nodes only go up by 10.

I also like the idea that only roundstart gamemode blobs would be able to make new cores, but what do you guys think about that. Perhaps a sprite change between the roundstart ones and created ones.

File: code/game/gamemodes/blob/powers.dm (hunk 1 above)
Comment (ghost) [reply code/game/gamemodes/blob/powers.dm]: thanks_sood

**Comment by ghost** [comment]
@clusterfack It doesn't give a name as to whodunnit, just 'you feel an echo resound through your mind... ONE DAY WHILE ANDY WAS MASTURBATING, WOODY GOT WOOD. HE COULD NO LONGER HELP HIMSEL...'

Upscaling the node amount I can do. 10 nodes for 1 blob does sound a tad much.

File: code/game/gamemodes/blob/powers.dm (hunk 1 above)
Comment (clusterfack) [inline code/game/gamemodes/blob/powers.dm]: They were telling you how to loop through world, not that you should

**Comment by clusterfack** [comment]
Something like 100, 125, 150, etc seems ok to me personally

**Comment by Kammerjunk** [comment]
Something linear like that would be nice. Currently, I think the _main_ issue is when a blob gets 10 free nodes during blobstorm events, but blob is still also massively overpowered during the actual gamemode.

@clusterfack I think Deity's last rework of blob did the "only roundstart can make cores" thing. It separated cores into two parts, overminds and cerebrates, and only overminds get the insane resource generation boost and the ability to spawn cores.

File: code/game/gamemodes/blob/powers.dm (hunk 1 above)
Comment (Kammerjunk) [reply code/game/gamemodes/blob/powers.dm]: I think the game currently does track the amount of cores in the game. It tells you the amount of cores in the Blob Status tab when you're a blob. I don't know how easy it'd be to work in here, though.

Comment (ghost) [reply code/game/gamemodes/blob/powers.dm]: i'll see what code that uses

Edit: Seems there's a global list of blob cores, I'll try and use that

**Comment by clusterfack** [comment]
Does it actually give those nodes to the blob overmind? I thought they were just freestanding nodes.

**Comment by ghost** [comment]
You put your viewing area over them, wait for them to pulse, then they're slaved to you so you can get magical wireless blob points

**Comment by Kammerjunk** [comment]
Well, the nodes don't give you resources, but they'll pulse any nearby resource nodes you put there. Resource nodes are the cheap ones at 40, nodes cost 60.

**Comment by clusterfack** [comment]
Is that really true? Its not explained anywhere, thats fuckin awful

**Comment by Kammerjunk** [comment]
Yup. Someone became the sole blob during blobstorm a few rounds ago and had like 10 resource blobs shortly after because of the nodes.

**Comment by clusterfack** [comment]
I think that either needs to be removed or made more obvious jeeze that is just awful

**Comment by ghost** [comment]
i remember deity Admin-PMing blobs about it, including yred

It's now how yred gets ahead when they get caught in a blob conglomerate, as every node is in a list you can jump to

**Comment by Allspice** [comment]
yep, rush 10 resources then 10 cores in 10 minutes

**Comment by clusterfack** [comment]
Jesus what a terrible idea, just remove it

**Comment by ghost** [comment]
Seems I may be mistaken

You can take control of a type of blob, but only resource nodes.

/obj/effect/blob/resource/Pulse(var/pulse = 0, var/origin_dir = 0)
    if(!overmind)
        var/mob/camera/blob/B = (locate() in range(src,1))
        if(B)
            to_chat(B,"<span class='notice'>You take control of the resource blob.</span>")
            overmind = B
            B.special_blobs += src
            B.update_specialblobs()
            update_icon()
    ..()

You can still jump between all nodes and place resource nodes next to them regardless of ownership

**Comment by Blithering** [comment]
Would be nice if blob could only expand from or benefit from sections that are attached to the central core.

**Comment by Exxion** [comment]
That would be good for a number of reasons

I was going to say coding it without it lagging might be difficult, but then I realized it could work with the existing pulses
It wouldn't update instantly, but it wouldn't lag any more than it already does and would also come with a handy visual reference of what's connected to you

**Comment by ghost** [comment]
I would love to change it to that, but I haven't a bloody clue how to do that.

File: code/game/gamemodes/blob/blobs/node.dm [hunk 2]
```diff
@@ -1,3 +1,5 @@
+#define BLOBPOINTINC 10
```
Comment (ghost) [inline code/game/gamemodes/blob/blobs/node.dm]: Fug

**Comment by Intigracy** [comment]
Blobs can't choose their own theme, right?

**Comment by ghost** [comment]
They can

**Comment by Intigracy** [comment]
I'd rather any gimmick ones like the AME require admin intervention personally

**Comment by clusterfack** [comment]
Make a global variable where sprites for blobs come from. Then make that variable editable from a limited set by the first blob in a round, and make a expanded version of the list available to admins so they can make meme blobs.

**Comment by Intigracy** [comment]
Admin verb:
https://github.com/d3athrow/vgstation13/blob/82403c717536aea5fecb9341bba9cef9d8948704/code/modules/admin/admin_verbs.dm#L1057

List of available looks:
https://github.com/d3athrow/vgstation13/blob/aa5e0b12db8bc9a31e1c618b0d046c9fb5650e7d/code/game/gamemodes/blob/theblob.dm#L281

Looks choosable by blobs:
https://github.com/d3athrow/vgstation13/blob/a559bcf4e9f3004ad5969e1344680db1f37575a2/code/game/gamemodes/blob/blobs/core.dm#L165

Already exists in every way you wanted, the ones he's adding just need to be restricted the same way the adminbus one is.

**Comment by ghost** [comment]
@Intigracy Thanks, I'll change that now

**Comment by Intigracy** [comment]
I suggest doing a blob_looks_locked or something list rather than doing a - for each one.

**Comment by ghost** [comment]
So, have 2 lists, one for blob looks players can choose, and another for admins to use?

**Comment by ghost** [comment]
Okay, the list is used in a whoooole lot more than just those 2 procs, and there's a lot of stuff that is specific list dependent

blob_looks

**Comment by Intigracy** [comment]
Second link should be
https://github.com/d3athrow/vgstation13/blob/82403c717536aea5fecb9341bba9cef9d8948704/code/modules/admin/admin_verbs.dm#L1057
, oops.

**Comment by Intigracy** [comment]
Yeah.

https://github.com/d3athrow/vgstation13/blob/aa5e0b12db8bc9a31e1c618b0d046c9fb5650e7d/code/game/gamemodes/blob/theblob.dm#L281
This one would be the ones selectable by players (so remove the adminbus
one).

Then another list below it with the admin selectable ones.

For admins all you have to do is add the second list to this proc a few
lines down (on mobile or I'd link the line)
https://github.com/d3athrow/vgstation13/blob/aa5e0b12db8bc9a31e1c618b0d046c9fb5650e7d/code/game/gamemodes/blob/theblob.dm#L281

**Comment by ghost** [comment]
I already did these things tho

It's the fact that the GUI buttons now all summon spores that's the problem now

**Comment by Intigracy** [comment]
Ignore those, I sent those in email reply days ago.

Jesus christ github get your shit together.

**Comment by clusterfack** [comment]
Congrats on finding your first byond bug. Setting nonconstant values using set name = "[something]" causes incorrect verbs to be called when manually calling verbs and causes them to not show up in the verb panel.

**Comment by clusterfack** [comment]
Also please remove only the pulsing animation, the one that gets sent from nodes and plays on all the blobs one after another until it reaches one it can act on.

**Comment by ghost** [comment]
Alrighty. Primary reason I commented out everything using blob_looks[looks] is because of the requested switch to the 2 lists rather than the one. 

I may have thought of a way round this however, working on that now.

File: code/modules/admin/admin.dm [hunk 3]
```diff
@@ -190,7 +190,8 @@ var/global/floorIsLava = 0
 				<A href='?src=\ref[src];simplemake=corgi;mob=\ref[M]'>Corgi</A> |
 				<A href='?src=\ref[src];simplemake=ian;mob=\ref[M]'>Ian</A> |
 				<A href='?src=\ref[src];simplemake=crab;mob=\ref[M]'>Crab</A> |
-				<A href='?src=\ref[src];simplemake=coffee;mob=\ref[M]'>Coffee</A>
+				<A href='?src=\ref[src];simplemake=coffee;mob=\ref[M]'>Coffee</A> |
+				<A href='?src=\ref[src];simplemake=blob;mob=\ref[M]'>BLOB</A>
```
Comment (ghost) [inline code/modules/admin/admin.dm:194]: Thank you so much for this

**Comment by ghost** [comment]
Changelog added, have I done that right?

**Comment by clusterfack** [comment]
Probably

**Comment by ghost** [comment]
I hate git

**Pull request closed without merging**